### PR TITLE
Move spender validation only for approvals

### LINKED ISSUE #7018: Owner Account Unable to Remove Allowance after Spender Account Deleted
### Description

If **Owner** account creates an allowance for **Spender** account for a fungible token and then the **Spender** account is deleted the **Owner** account cannot remove the **Spender**'s allowance.  The attempt returns INVALID_ALLOWANCE_SPENDER_ID.

### Steps to reproduce

1. **Owner** account creates an allowance for **Spender**
2. **Spender** account is deleted. (Note: this does not delete the allowance)
3. **Owner** account attempts to remove allowance by setting amount to 0
4. The attempt fails and the network returns `INVALID_ALLOWANCE_SPENDER_ID`

The _Expected_ behavior would be one of two things:

1. The allowance that was designated by **Owner** would be removed when **Spender** account was deleted (preferred)
2. The **Owner** account would be able to remove the **Spender** allowance without error.

### Additional context

_No response_

### Hedera network

testnet, previewnet

### Version

.net SDK 10.0.0 beta 7

### Operating system

Windows

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #7018 

Currently spender validation exist in `CryptoApproveAllowance` logic. But there is no need to validate spender exists when amount == 0 which is deleting allowance. 
Instead need to delete the allowance if it exists. This is necessary because spender could be deleted by the time allowance is being removed by the owner.

### CODE REVIEW

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java [hunk 1]
```diff
@@ -173,6 +178,7 @@ protected void applyNftAllowances(final List<NftAllowance> nftAllowances, final
             final var owner = allowance.getOwner();
             final var approvingAccount = fetchOwnerAccount(owner, payerAccount, accountStore, entitiesChanged);
             final var spenderId = Id.fromGrpcAccount(allowance.getSpender());
+            // To add allowances spender should be validated as being a valid account
```
Comment (tinker-michaelj) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java:181]: Following your pattern above, wouldn't we want to move this `loadAccountOrFailWith()` down before L#189,
```
  ...
                final var key = FcTokenAllowanceId.from(tokenId.asEntityNum(), spenderId.asEntityNum());
                if (allowance.getApprovedForAll().getValue()) {
                    // We skip this validation below when revoking an operator approval
                    accountStore.loadAccountOrFailWith(spenderId, INVALID_ALLOWANCE_SPENDER_ID);
                    approveForAllNfts.add(key);
                } else {
                    approveForAllNfts.remove(key);
                }
```

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java [hunk 2]
```diff
@@ -101,14 +101,16 @@ private void applyCryptoAllowances(final List<CryptoAllowance> cryptoAllowances,
             final var cryptoMap = accountToApprove.getMutableCryptoAllowances();
 
             final var spender = Id.fromGrpcAccount(allowance.getSpender());
-            accountStore.loadAccountOrFailWith(spender, INVALID_ALLOWANCE_SPENDER_ID);
-
             final var amount = allowance.getAmount();
 
             if (cryptoMap.containsKey(spender.asEntityNum()) && amount == 0) {
+                // spender need to be validated as being a valid account when removing allowances,
```
Comment (tinker-michaelj) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java:107]: `// spender need NOT be validated...` here?

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java:107]: ah yes 😞

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java [hunk 3]
```diff
@@ -140,16 +142,19 @@ private void applyFungibleTokenAllowances(final List<TokenAllowance> tokenAllowa
...
 
             final var spender = Id.fromGrpcAccount(allowance.getSpender());
-            accountStore.loadAccountOrFailWith(spender, INVALID_ALLOWANCE_SPENDER_ID);
-
             final var amount = allowance.getAmount();
             final var tokenId = allowance.getTokenId();
 
             final var key = FcTokenAllowanceId.from(EntityNum.fromTokenId(tokenId), spender.asEntityNum());
             if (tokensMap.containsKey(key) && amount == 0) {
+                // spender need to be validated as being a valid account when removing allowances,
```
Comment (mhess-swl) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java:150]: Is this also needed here?
`// spender need NOT be validated...`

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java:150]: yes fixed.

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java]: Yes Michael, fixed. Added that on line 189 and also when `serialsCount > 0`. Thanks for pointing out 🙌

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java [hunk 4]
```diff
@@ -173,20 +178,27 @@ protected void applyNftAllowances(final List<NftAllowance> nftAllowances, final
...
+                    // validated as being a valid account
+                    accountStore.loadAccountOrFailWith(spenderId, INVALID_ALLOWANCE_SPENDER_ID);
                     approveForAllNfts.add(key);
                 } else {
                     approveForAllNfts.remove(key);
                 }
                 validateAllowanceLimitsOn(approvingAccount, dynamicProperties.maxAllowanceLimitPerAccount());
             }
 
+            if (allowance.getSerialNumbersCount() > 0) {
```
Comment (tinker-michaelj) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/ApproveAllowanceLogic.java:197]: Ah yes, nice catch! I missed this. 🥂

**Review by tinker-michaelj — DISMISSED** [review]
LGTM!

**Review by mhess-swl — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Pull request merged**